pkg/dao/rpm.go

```diff
@@ -417,3 +429,47 @@ func FilteredConvert(yumPkgs []yum.Package, excludeChecksums []string) []models.
 	}
 	return dbPkgs
 }
+
+func convertRPMSortByToSQL(SortBy string) string {
+	sqlOrderBy := ""
+
+	sortByArray := strings.Split(SortBy, ",")
+	lengthOfSortByParams := len(sortByArray)
+
+	sortMap := map[string]string{
+		"name":    "name",
+		"release": "release",
+		"version": "version",
+		"arch":    "arch",
+	}
+
+	for i := 0; i < lengthOfSortByParams; i++ {
+		sortBy := sortByArray[i]
+
+		split := strings.Split(sortBy, ":")
+		ascOrDesc := " asc"
+
+		if len(split) > 1 && split[1] == "desc" {
+			ascOrDesc = " desc"
+		}
+
+		sortField, ok := sortMap[split[0]]
+
+		// Only update if the sortMap above returns a valid value
+		if ok {
+			// Concatenate (e.g. "url desc," + "name" + " asc")
+			sqlOrderBy = sqlOrderBy + sortField + ascOrDesc
+
+			// Add a comma if this isn't the last item in the "sortByArray".
+			if i+1 < lengthOfSortByParams {
+				sqlOrderBy = sqlOrderBy + ", "
+			}
+		}
+	}
```

Comment:
can we pull most of this to a helper function that could also be used by  dao/repository_config.go?

Reply:
I was thinking this too, but where would that helper function live?

Reply:
I don't think there's a precedent for a function shared between two dao files (except in tests). Maybe a `dao/helpers.go` file?

Reply:
works4me!